TWLight/users/models.py

```diff
@@ -247,46 +362,29 @@ def get_wp_groups_display(self):
         else:
             return None
 
-    def _is_user_valid(self, identity, global_userinfo):
-        """
-        Check for the eligibility criteria laid out in the terms of service.
-        To wit, users must:
-        * Have >= 500 edits
-        * Be active for >= 6 months
-        * Not be blocked on any projects
-
-        Note that we won't prohibit signups or applications on this basis.
-        Coordinators have discretion to approve people who are near the cutoff.
-        """
-        # If, for some reason, this information hasn't come through,
-        # default to user not being valid.
-        if not global_userinfo:
-            return False
-        # Check: >= 500 edits
-        enough_edits = int(global_userinfo["editcount"]) >= 500
-
-        # Check: registered >= 6 months ago
-        # Try oauth registration date first. If it's not valid,
-        # try the global_userinfo date
-        try:
-            reg_date = datetime.strptime(identity["registered"], "%Y%m%d%H%M%S").date()
-        except:
-            reg_date = datetime.strptime(
-                global_userinfo["registration"], "%Y-%m-%dT%H:%M:%SZ"
-            ).date()
-        account_old_enough = datetime.today().date() - timedelta(days=182) >= reg_date
-
-        # Check: not blocked
-        not_blocked = identity["blocked"] == False
-
-        if enough_edits and account_old_enough and not_blocked:
+    def bundle_eligible(self):
+        if self.wp_valid and self.wp_enough_recent_edits:
             return True
         else:
-            logger.info(
-                "Editor {username} was not valid.".format(username=self.wp_username)
-            )
             return False
 
+    def _set_recent_edits(self):
+        # If it's been 30 days or more since we last updated historical editcount,
+        # copy wp_editcount to wp_editcount_prev before updating wp_editcount.
+        # This allows us track recent edits for bundle eligibility.
+        if self.wp_editcount_prev_date_updated:
+            editcount_update_delta = date.today() - self.wp_editcount_prev_date_updated
+            if editcount_update_delta.days >= 30:
+                self.wp_editcount_prev = self.wp_editcount
+                self.wp_editcount_prev_date_updated = date.today()
+                recent_edits = self.wp_editcount - self.wp_editcount_prev
+                if recent_edits >= 10:
+                    self.wp_enough_recent_edits = True
+                else:
+                    self.wp_enough_recent_edits = False
+        else:
+            self.wp_editcount_prev_date_updated = date.today()
+
```

Comment:
Don't we want to always be updating `wp_enough_recent_edits` on login so that we can check the most up-to-date edit count? In that way an editor who doesn't qualify can go make enough edits, then come back and upon logging in be flagged as eligible.

Reply:
This PR is still a work in progress and isn't ready for review. It's just now to the point where it doesn't break existing code. This block is going to check either 30 days or the required delta in edit count

Reply:
Sure! Was just browsing through and wanted to make sure this behaviour didn't become too difficult to disentangle down the line :)